apps/doji-frontend/src/pages/schedule-session/index.tsx

```diff
@@ -0,0 +1,268 @@
+import ConfirmDialog from '@frontend/components/ExpertService/ConfirmDialog'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { IScheduleSessionDTO, IServiceInformationDTO } from '@libs/api'
+import { SearchBar, Tables, TopBar, TopBarActionType } from '@libs/mui'
+import { DatePicker, TimePicker } from '@mui/lab'
+import AdapterDateFns from '@mui/lab/AdapterDateFns'
+import LocalizationProvider from '@mui/lab/LocalizationProvider'
+import { Avatar, Box, Button, Container, Grid, TextField, Typography } from '@mui/material'
+import axios from 'axios'
+import * as yup from 'yup'
+
+import React, { useEffect } from 'react'
+import { Controller, SubmitHandler, useForm } from 'react-hook-form'
+
+import TagsInput from '../../components/ExpertService/TagInput'
+
+const today = new Date()
+today.setHours(0, 0, 0, 0)
+const createScheduleValidation = yup.object({
+  date: yup
+    .date()
+    .typeError('This date is invalid')
+    .required('Please enter the date.')
+    .min(today, 'The date is invalid'),
+  startTime: yup
+    .date()
+    .typeError('The start time is invalid')
+    .required('Please enter the start time'),
+  endTime: yup
+    .date()
+    .typeError('The end time is invalid')
+    .required('Please enter the start end')
+    .min(yup.ref('startTime'), 'End time must be after start time'),
+  participants: yup.array(),
+})
+
+type scheduleModel = yup.InferType<typeof createScheduleValidation>
+
+export function Index() {
+  const [openDialog, setOpenDialog] = React.useState(false)
+  const [confirm, setConfirm] = React.useState(false)
+  const [serviceData, setServiceData] = React.useState<IServiceInformationDTO>(null)
+  const [scheduleSessionData, setscheduleSessionData] = React.useState<IScheduleSessionDTO>({
+    fee: 0,
+    expertUsername: '',
+    serviceName: '',
+    duration: 0,
+    startTime: new Date(),
+    participantsUsername: [],
+  })
+  const {
+    register,
+    watch,
+    handleSubmit,
+    formState: { errors },
+    control,
+  } = useForm<scheduleModel>({
+    resolver: yupResolver(createScheduleValidation),
+    defaultValues: {
+      date: new Date(),
+      startTime: new Date(),
+      endTime: new Date(),
+      participants: [],
+    },
+  })
+  const watchAll = watch()
```

Comment:
dont do this because this will cause the worst performance (HUGH number of unnecessary re-rendering)

Reply:
use `useWatch` instead